enhancements/ingress/default-apps-domain.md

```diff
@@ -0,0 +1,118 @@
+---
+title: default-apps-domain
+authors:
+  - "@dustman9000"
+reviewers:
+  - "@Miciah"
+  - "@danehans"
+  - "@frobware"
+  - "@knobunc"
+  - "@sgreene570"
+approvers:
+  - "@knobunc"
+  - "@Miciah"
+creation-date: 2020-08-12
+last-updated: 2020-09-04
+status: implementable
+---
+
+# Default Apps Domain
+
+## Release Signoff Checklist
+
+- [X] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+This enhancement allows for the configuration of a default apps domain in which
+can be used to set the default host and domain of the route, if not specified by
+the admin. Instead of using the default cluster domain, this new setting can be
+used to override it when set.
+
+## Motivation
+
+Many customers have requested "Custom Domains" in which they would like an already
+installed Openshift cluster to support additional domains for their apps by default.
+This domain is owned by a customer in which they would point a wildcard CNAME record
+to an existing well known CNAME record.
+
+### Goals
+
+1. Enable the cluster administor to specify the default apps domain seperate
+from the installed cluster domain.
+
+## Proposal
+
+The Ingress API is extended by adding an optional `AppsDomain` field with type
+`string`.
+
+```go
+type IngressSpec struct {
+    ...
+    // optional domain to be used instead of Domain installed cluster domain
+    // when Route Subdomain is unspecified.
+    // This assumes a new ingresscontroller has been setup with a wildcard
+    // certificate.
+    // +optional
+    AppsDomain string `json:"appsDomain,omitempty"`
+}
+```
+
+The following example configures apps.acme.io as the default apps domain.
+
+```yaml
+apiVersion: config.openshift.io/v1
+kind: Ingress
+metadata:
+  name: cluster
+spec:
+  domain: apps.drow-dev01.w9h5.s1.openshiftapps.com
+  appsDomain: apps.acme.io
+```
+
+### Implementation Details
+
+Implementation of this enhancement requires changes in the following repositories:
+
+* openshift/api
+* openshift/cluster-openshift-apiserver-operator
+
+In [config/v1/types_ingress.go](https://github.com/openshift/api/blob/master/config/v1/types_ingress.go) of [openshift/api](https://github.com/openshift/api), add new field: `appsDomain`.
+In [observe_ingresses.go](https://github.com/openshift/cluster-openshift-apiserver-operator/blob/master/pkg/operator/configobservation/ingresses/observe_ingresses.go) of [openshift/cluster-openshift-apiserver-operator](https://github.com/openshift/cluster-openshift-apiserver-operator), add logic to check if `appsDomain` is set. If that field is set use it insead of Spec.Domain for the value of `routingDomain`.
```

Comment:
```suggestion
In [observe_ingresses.go](https://github.com/openshift/cluster-openshift-apiserver-operator/blob/master/pkg/operator/configobservation/ingresses/observe_ingresses.go) of [openshift/cluster-openshift-apiserver-operator](https://github.com/openshift/cluster-openshift-apiserver-operator), add logic to check if `appsDomain` is set. If that field is set, use it instead of `spec.Domain` for the value of `routingDomain`.
```

Reply:
Yes, that's what I meant. Thanks.